packages/block-library/src/group/deprecated.js

```diff
@@ -109,15 +109,11 @@ const deprecated = [
 			);
 		},
 		isEligible: ( { layout } ) =>
-			! layout ||
-			layout.inherit ||
-			( layout.contentSize && layout.type !== 'constrained' ),
+			layout?.inherit ||
+			( layout?.contentSize && layout?.type !== 'constrained' ),
```

Comment:
This fix makes sense. The deprecation is eligible when only experimental layout values are present and not when layout values are missing.

Ref: https://github.com/WordPress/gutenberg/pull/57544#issuecomment-1940336043